agent/dns.go

```diff
@@ -1899,9 +1899,13 @@ func (d *DNSServer) resolveCNAME(cfg *dnsConfig, name string, maxRecursionLevel
 		resp := &dns.Msg{}
 
 		req.SetQuestion(name, dns.TypeANY)
-		// TODO: handle error response
-		d.dispatch(nil, req, resp, maxRecursionLevel-1)
 
+		err := d.dispatch(nil, req, resp, maxRecursionLevel-1)
+		rCode := rCodeFromError(err)
+		if rCode == dns.RcodeNameError || errors.Is(err, errNoAnswer) {
+			d.addSOA(cfg, resp)
+		}
+		resp.SetRcode(req, rCode)
 		return resp.Answer
```

Comment:
@dhiaayachi  I think  the challenge with this TODO is that we only return `resp.Answer`.

`SetRcode and `addSOA` change different parts of the `resp`, they don't modify the `Answer` section, so when we return `Answer` here we throw away the `Rcode` and `NS` sections that we modified.

My TODO comment was not very complete, but I think the question we need to answer is: Do we need to care about the  error returned from `dispatch` ?

If we need to care, then `resolveCNAME` needs to return either the full `resp` or some other struct to include all of that information.

Reply:
Another possibility is that `dispatch` is just the wrong abstraction here, but if that's the case it'll be more work to clean up I think.

Reply:
yes you are right, I also when through the call flow and we don't really care about dispatch failing. it end up returning a nil answer which mean that at the end we won't add any extra fields.